Feed refresh not working properly

**App version:** 1.5.2.0 (from Google Play), already had the issue in 1.4.2.2

**Android version**: 6.1

**Device model**: LG Nexus 5X

**Expected behaviour**:
On the "Episodes" Screen I click on the refresh button to check whether new episodes of my podcasts are available. In the status bar Antenna Pod shows a shrinking number of how many feeds still must be updated. After all feeds are updated it shows the message "processing downloads" and after this is closed all new episodes are shown in the "Episodes" Screen.

**Current behaviour**:
- My feeds are not updated with this refresh function. Nevertheless an update on each feed still works.
- The message "processing downloads" persists. It seems to do something in the background all the time, because my battery is drained. The only chance to get rid of this is to kill Antenna Pod.

**First occured**: Version 1.4.2.2 / about 4-6 weeks ago/
I'm using Antenna Pod since beginning of year 2016. I already had the same problem in the end of January. I reinstalled Antenna Pod (which fixed the problem). About 4 weeks later, it appeared again. A week later I had no problem anymore, now it appears again.

**Steps to reproduce**:
It happens every time I update my feeds.
Before it happened the first time, this issue did not occure. I didn't change settings in-between.

**Environment**:
I'm using Prestissimo.
No Auto-download, Auto-update feeds once a day.

**Stacktrace/Logcat**:
The way it described in this wiki, I couldn't get it to work on my phone.

Brilliant, I think I found the issue:
```
W/System.err: java.util.concurrent.ExecutionException: java.lang.NullPointerException
W/System.err:     at java.util.concurrent.FutureTask.report(FutureTask.java:93)
W/System.err:     at java.util.concurrent.FutureTask.get(FutureTask.java:163)
W/System.err:     at de.danoeh.antennapod.core.service.download.DownloadService$FeedSyncThread.run(DownloadService.java:753)
W/System.err: Caused by: java.lang.NullPointerException
W/System.err:     at de.danoeh.antennapod.core.feed.FeedFile.compareWithOther(FeedFile.java:57)
W/System.err:     at de.danoeh.antennapod.core.feed.FeedMedia.compareWithOther(FeedMedia.java:185)
W/System.err:     at de.danoeh.antennapod.core.feed.FeedItem.updateFromOther(FeedItem.java:199)
W/System.err:     at de.danoeh.antennapod.core.storage.DBTasks.updateFeed(DBTasks.java:596)
W/System.err:     at de.danoeh.antennapod.core.service.download.DownloadService$FeedSyncThread$1.run(DownloadService.java:764)
W/System.err:     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
W/System.err:     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
W/System.err:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
```

--- a/core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java
@@ -324,7 +324,7 @@ public class Feed extends FeedFile implements FlattrThing, ImageResource {
         if (super.compareWithOther(other)) {
             return true;
         }
-        if (!title.equals(other.title)) {
+        if (!TextUtils.equals(title, other.title)) {
             return true;
         }
         if (other.feedIdentifier != null) {

--- a/core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/feed/Feed.java
@@ -328,8 +328,7 @@ public class Feed extends FeedFile implements FlattrThing, ImageResource {
             return true;
         }
         if (other.feedIdentifier != null) {
-            if (feedIdentifier == null
-                    || !feedIdentifier.equals(other.feedIdentifier)) {
+            if (feedIdentifier == null || !feedIdentifier.equals(other.feedIdentifier)) {
                 return true;
             }
         }

--- a/core/src/main/java/de/danoeh/antennapod/core/feed/FeedFile.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/feed/FeedFile.java
@@ -1,5 +1,7 @@
 package de.danoeh.antennapod.core.feed;
 
+import android.text.TextUtils;
+
 import java.io.File;
 
 /**
@@ -54,7 +56,7 @@ public abstract class FeedFile extends FeedComponent {
         if (super.compareWithOther(other)) {
             return true;
         }
-        if (!download_url.equals(other.download_url)) {
+        if (!TextUtils.equals(download_url, other.download_url)) {
             return true;
         }
         return false;

--- a/core/src/main/java/de/danoeh/antennapod/core/feed/FeedPreferences.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/feed/FeedPreferences.java
@@ -95,8 +95,9 @@ public class FeedPreferences {
      * @return True if the two objects are different.
      */
     public boolean compareWithOther(FeedPreferences other) {
-        if (other == null)
+        if (other == null) {
             return true;
+        }
         if (!TextUtils.equals(username, other.username)) {
             return true;
         }

